src/components/textarea/textarea.pw.tsx

```diff
@@ -0,0 +1,1070 @@
+
+    const inputElementParentParent = getDataElementByValue(page, "input")
+      .locator("..")
+      .locator("..");
+
+    await expect(inputElementParentParent).toHaveCSS("max-width", "100%");
+  });
+
+  test("should render with name prop", async ({ mount, page }) => {
+    await mount(<TextareaComponent name={CHARACTERS.STANDARD} />);
+
+    await expect(textareaChildren(page)).toHaveAttribute(
+      "name",
+      CHARACTERS.STANDARD
+    );
+  });
+
+  test("should render with disabled prop", async ({ mount, page }) => {
+    await mount(<TextareaComponent disabled />);
+
+    const textareaChildrenElement = textareaChildren(page);
+
+    await expect(textareaChildrenElement).toBeDisabled();
+  });
+
+  test("should render icon with disabled style", async ({ mount, page }) => {
+    await mount(<TextareaComponent disabled inputIcon="bin" />);
+
+    const textareaIcon = textarea(page).locator(ICON);
+
+    await expect(textareaIcon).toBeVisible();
+    await expect(textareaIcon).toHaveCSS("color", "rgba(0, 0, 0, 0.3)");
+  });
+
+  testData.forEach((placeholder) => {
+    test(`should render with placeholder prop set to ${placeholder}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<TextareaComponent placeholder={placeholder} />);
+
+      await expect(textareaChildren(page)).toHaveAttribute(
+        "placeholder",
+        placeholder
+      );
+    });
+  });
+
+  test("should render with autoFocus prop and correct styling when focusRedesignOptOut is false", async ({
+    mount,
+    page,
+  }) => {
+    await mount<HooksConfig>(<TextareaComponent autoFocus />, {
+      hooksConfig: {
+        focusRedesignOptOut: false,
+      },
+    });
+
+    const textareaChildrenElement = textareaChildren(page);
+    await expect(textareaChildrenElement).toBeFocused();
+
+    await expect(textarea(page)).toHaveCSS(
+      "box-shadow",
+      "rgb(255, 188, 25) 0px 0px 0px 3px, rgba(0, 0, 0, 0.9) 0px 0px 0px 6px"
+    );
+  });
+
+  test("should render with autoFocus prop and correct styling when focusRedesignOptOut is true", async ({
+    mount,
+    page,
+  }) => {
+    await mount<HooksConfig>(<TextareaComponent autoFocus />, {
+      hooksConfig: {
+        focusRedesignOptOut: true,
+      },
+    });
+
+    await expect(textareaChildren(page)).toBeFocused();
+    await checkGoldenOutline(textarea(page));
+  });
+
+  test("should render with readOnly prop", async ({ mount, page }) => {
+    await mount(<TextareaComponent readOnly />);
+
+    await expect(textareaChildren(page)).not.toBeEditable();
+  });
+
+  test("should render icon with readOnly style", async ({ mount, page }) => {
+    await mount(<TextareaComponent readOnly inputIcon="bin" />);
+
+    const textareaIcon = textarea(page).locator(ICON);
+    await expect(textareaIcon).toBeVisible();
+    await expect(textareaIcon).toHaveCSS("color", "rgba(0, 0, 0, 0.3)");
+  });
+
+  (["error", "warning", "info"] as const).forEach((type) => {
```

Comment:
**comment** (non-blocking): this *might* be another `as const` that we can get rid of. I saw "might", just because my VS Code doesn't seem to complain when I remove it, but I'm not actually sure why - spreading `{ [type]: message }` on to the component when `type` is just inferred as any old `string` ought to be an error imo. Maybe we should keep this one as `as const` even if it is valid to remove it, but anyway I thought I'd mention it and will leave it up to you...